ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,270 @@
+                {
+                    Compression.Zip(dataFile, deleteOriginal: true);
+                    Log.Trace($"PsychSignalDataConverter.CompressData(): Successfully compressed: {dataFile}");
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Handle to a file so that we don't have to open and close it every time we want
+        /// to write to a file. This helps us speed up time spent processing massively.
+        /// </summary>
+        private class TickerData : IDisposable
+        {
+            private readonly DirectoryInfo _destinationDirectory;
+            private readonly string _ticker;
+
+            private StreamWriter _writer;
+            private string _tempPath;
+            private DateTime _date;
+
+            /// <summary>
+            /// Creates writer instances and saves file path.
+            /// Used to keep file open until the path changes for the symbol
+            /// </summary>
+            /// <param name="ticker"></param>
+            /// <param name="date"></param>
+            /// <param name="destinationDirectory"></param>
+            public TickerData(string ticker, DateTime date, DirectoryInfo destinationDirectory)
+            {
+                _date = date;
+                _ticker = ticker;
+                // Using win32 Path.GetTempFileName can cause filename collisions
+                _tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                _writer = new StreamWriter(_tempPath);
+                _destinationDirectory = destinationDirectory;
+            }
+            
+            /// <summary>
+            /// Adds a new line to the writer
+            /// </summary>
+            /// <param name="timestamp">Event timestamp in UTC</param>
+            /// <param name="csv">CSV enumerable</param>
+            public void Append(DateTime timestamp, IEnumerable<string> csv)
+            {
+                var date = timestamp.Date;
+
+                if (date != _date)
+                {
+                    Dispose();
+
+                    _date = date;
+                    _tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                    _writer = new StreamWriter(_tempPath);
+                }
+
+                _writer.WriteLine(ToCsv(timestamp, csv.Skip(3)));
+            }
+            
+            /// <summary>
+            /// Moves the temporary file containing data to the final path, deleting
+            /// any existing file to avoid conflicts when moving
+            /// </summary>
+            private void MoveTempFile()
+            {
+                var tickerDirectory = Path.Combine(_destinationDirectory.FullName, _ticker);
+                var writePath = Path.Combine(tickerDirectory, $"{_date:yyyyMMdd}.csv");
+
+                Directory.CreateDirectory(tickerDirectory);
```

Comment:
Maybe to reduce effort the `tickerDirectory` could be created at construction time?

Reply:
another option is to create in a static constructor, or provide a static flag so we're not creating it every time we create a new base data type.